### PR TITLE
Issue223 write unit tests

### PR DESCRIPTION
**Pull request by lmcarboneau**
This PR should address #111 and some parts of #223 together, since they're both testing related. 
Screenshot from 2020-05-22 08-48-11

The owner of the repo (@grd349) will need to authorize Travis CI on the repo by [signing in to their website with Github](https://travis-ci.com/signin) and accepting the authorization request. It will authorize it to test any of your repos, so you may want to select only those you want to include testing for in the Travis CI

### CODE REVIEW

**Comment by grd349** [comment]
@lmcarboneau Hey Lindsey,

I think I sorted out the travis CI (that could be a lie).  Is there something else I need to do?

**Comment by lmcarboneau** [comment]
I think it should be ready to test, I didn't realize it was still marked WIP. I'll open it for review :)

**Comment by lmcarboneau** [comment]
@grd349 Hi Guy,
It's possible this should be merged into a `devel` branch of some kind first, instead of straight into master - mostly it depends on how much you care about the tests not starting and running automatically on the first commit. If there's an issue, it should only be on the TravisCI end, but it might mean that testing is unavailable on the master branch while it's being sorted. It would have to be done eventually though so... I'm not sure what the right answer is  😅

In short, it should be safe to merge, but there might be some settings that need sorting if the tests don't run.
